src/function/onlyWhen/onlyWhen.test.ts

```diff
@@ -0,0 +1,20 @@
+import * as assert from 'assert'
+
+import { id } from '..'
+import { onlyWhen } from './onlyWhen'
+
+describe(`onlyWhen`, () => {
+  describe(`given a predicate that is false`, () => {
+    it(`returns default value`, () => {
+      const expected = 0
+      assert.strictEqual(expected, onlyWhen((_: number) => false, expected, id, 1))
```

Comment:
ah, actual then expected.  I'd gotten used to putting expected first with https://github.com/briancavalier/assert.  Thanks, will fix.